File: app/src/main/java/co/atrasvida/example/ApiClient/ApiConfig.java
package co.atrasvida.example.ApiClient;

import java.lang.reflect.Type;

import co.atrasvida.avidawebapi_annotations.WebApiConfig;

public class ApiConfig extends WebApiConfig {

    @Override
    public String getBaseUrl() {
        return URL.BASE_URL;
    }

    @Override
    public Type getBaseModel() {
        return BaseResponse.class;
    }

    @Override
    public String getToken() {
        return "Bearer $token";
    }

    @Override
    public long getReadTimeout() {
        return 20;
    }

    @Override
    public long getConnectTimeout() {
        return 20;
    }

    @Override
    public long getWriteTimeout() {
        return 20;
    }

    @Override
    public Boolean isDebugMode() {
        return false;
    }


}

File: app/src/main/java/co/atrasvida/example/apiClient/ApiConfig.kt
package co.atrasvida.example.apiClient

import co.atrasvida.avidawebapi.BuildConfig
import co.atrasvida.avidawebapi_annotations.WebApiConfig

class ApiConfig : WebApiConfig() {
    override fun getBaseUrl()= URL.BASE_URL

    override fun getBaseModel()= BaseResponse::class.java

    override fun getToken()=  null //"Bearer \$token"

    override fun getReadTimeout()=  20L

    override fun getConnectTimeout()= 20L

    override fun getWriteTimeout()= 20L

    override fun isDebugMode()= BuildConfig.DEBUG
}